arraykit.c

```diff
@@ -132,6 +137,108 @@ name_filter(PyObject *Py_UNUSED(m), PyObject *n)
     return n;
 }
 
+
+static PyObject *
+shape_filter(PyObject *Py_UNUSED(m), PyObject *a)
+{
+    // Represent a 1D array as a 2D array with length as rows of a single-column array.
+    // https://numpy.org/doc/stable/reference/c-api/array.html
+    // https://stackoverflow.com/questions/56182259/how-does-one-acces-numpy-multidimensionnal-array-in-c-extensions
```

Comment:
Convention is to put "docstring"-style comments like this on the line immediately above the definition (with no vertical whitespace).

It's a bit weird, but following conventions has benefits... for example, VSCode's C/C++ plugin will now display this comment as part of the hover text for this function!

```suggestion
// Represent a 1D array as a 2D array with length as rows of a single-column array.
// https://numpy.org/doc/stable/reference/c-api/array.html
// https://stackoverflow.com/questions/56182259/how-does-one-acces-numpy-multidimensionnal-array-in-c-extensions
static PyObject *
shape_filter(PyObject *Py_UNUSED(m), PyObject *a)
{
```

Reply:
Same goes for other names too, by the way! If you put a comment on the line *above* a variable or member declaration, VSCode will show you that comment when you hover over the name elsewhere!

(I didn't bother moving comments referring to variables, but you can if you want.)

Reply:
Got it, thanks.